modules/generator/storage/instance.go

```diff
@@ -98,6 +119,44 @@ func (s *storageImpl) Close() error {
 	).Err()
 }
 
+func (s *storageImpl) watchOverrides() {
```

Comment:
we already watch overrides at the generator/instance.go level. does it make sense to reload these overrides there as well? or does this need to be done here?

Reply:
Either approach works.

IMO, separating overrides updates per pkg results for cleaner code. If we moved this logic to `generator/instance.go` we would leak storage specific details outside `/generator/storage` and still would have dedicated code in this file in case of a required update.